code/game/objects/items/spear.dm

```diff
@@ -287,3 +287,114 @@
 	)
+
+/**
+ * Skybulge
+ *
+ * Gives a special ability that allows you to enter the skies an drop down upon a target.
+ * Other than that ability, is a default spear with extra throw force, but no embedding.
+ */
+/obj/item/spear/skybulge
+	name = "\improper Sky Bulge"
+	desc = "A legendary stick with a very pointy tip. Takes you to the skies!"
+	icon_state = "sky_bulge0"
+	icon_prefix = "sky_bulge"
+	attack_verb_continuous = list("attacks", "pokes", "jabs", "tears", "gores", "lances")
+	attack_verb_simple = list("attack", "poke", "jab", "tear", "gore", "lance")
+	slot_flags = parent_type::slot_flags | ITEM_SLOT_HANDS //this is needed for action types to give actions in-hand
+	throwforce = 24
+	embed_type = null //no embedding
+
+	custom_materials = list(
+		/datum/material/diamond = HALF_SHEET_MATERIAL_AMOUNT,
+		/datum/material/alloy/plastitaniumglass = SHEET_MATERIAL_AMOUNT,
+	)
+	actions_types = list(/datum/action/item_action/skybulge)
+
+///The action button the spear gives, usable once a minute.
+/datum/action/item_action/skybulge
+	name = "Dragoon Jump"
+	desc = "Jump up into the skies and fall down upon your opponents to deal double damage."
+	///Cooldown time between jumps.
+	var/jump_cooldown_time = 1 MINUTES
+	/**
+	 * boolean we set every time we jump, to know if we should take away the passflags we give,
+	 * so we don't give/take when they have it from other sources (since we don't have traits, we have
+	 * no way to tell which pass flags they get from what source.)
+	 */
+	var/gave_pass_flags = FALSE
+
+/datum/action/item_action/skybulge/do_effect(trigger_flags)
+	if(!HAS_TRAIT(target, TRAIT_WIELDED))
+		owner.balloon_alert(owner, "not dual-wielded!")
+		return
+	var/time_left = S_TIMER_COOLDOWN_TIMELEFT(target, COOLDOWN_SKYBULGE_JUMP)
+	if(time_left)
+		owner.balloon_alert(owner, "[FLOOR(time_left * 0.1, 0.1)]s cooldown!")
+		return
+	//do after shows the progress bar as feedback, so nothing here.
+	if(LAZYACCESS(owner.do_afters, target))
+		return
+
+	owner.balloon_alert(owner, "charging up...")
+	ADD_TRAIT(target, TRAIT_NEEDS_TWO_HANDS, ACTION_TRAIT)
+	INVOKE_ASYNC(src, PROC_REF(jump_up))
+
+///Sends the owner up in the air and calls them back down, calling land() for aftereffects.
+/datum/action/item_action/skybulge/proc/jump_up()
+	if(!do_after(owner, 2 SECONDS, target = owner, timed_action_flags = IGNORE_USER_LOC_CHANGE))
+		REMOVE_TRAIT(target, TRAIT_NEEDS_TWO_HANDS, ACTION_TRAIT)
+		return
+	playsound(owner, 'sound/effects/footstep/heavy1.ogg', 50, 1)
+	S_TIMER_COOLDOWN_START(target, COOLDOWN_SKYBULGE_JUMP, jump_cooldown_time)
+	new /obj/effect/temp_visual/telegraphing/exclamation/following(get_turf(owner), 2.5 SECONDS, owner)
+
+	RegisterSignal(target, COMSIG_ITEM_ATTACK, PROC_REF(on_attack_during_jump))
+	ADD_TRAIT(target, TRAIT_NODROP, ACTION_TRAIT)
+	ADD_TRAIT(owner, TRAIT_SILENT_FOOTSTEPS, ACTION_TRAIT)
+	owner.resistance_flags |= INDESTRUCTIBLE //not while jumping at least
```

Comment:
Wait I just noticed I think you are adding this to the mob, not the item?
`resistance_flags` are for object/structure damage as far as I know

Reply:
oh are they? that's a little weird

Reply:
yeah, you probably want TRAIT_GODMODE for this then